### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by MarcAntoine-Arnaud**

### CODE REVIEW

**Comment by greyblake** [comment]
Hi, @MarcAntoine-Arnaud !
Thanks for the pull request. These days I am extremally busy, I hope I'll be able to take a deep look at the code changes on weekend.

It looks like your PR has much more than update of the dependencies. Can you please the changes and the motivation behind them?

I'd like also to point, that the project is still in development... And number of things may be rearranged in the future.

**Comment by MarcAntoine-Arnaud** [comment]
Hi @greyblake ,

No problem for the delay ;-)

For dependencies, I have build a worker which use serde 1.0.9 version. And I need new features from this release.
As I need to get data via jsonpath too, I updated dependencies here, which was not complex.

I start also to add the json path filtering option.
I'm not sure to do the best right thing, so feel free to comment ;-)

Best,
Marc-Antoine

**Comment by greyblake** [comment]
@MarcAntoine-Arnaud Sorry, couldn't take a look this weekend. But it's on my TODO list.

File: src/structs.rs [hunk 1]
```diff
@@ -1,25 +1,42 @@
...
     // [10:20]
     Slice(::std::ops::Range<usize>),
-
     // [:7]
     SliceTo(::std::ops::RangeTo<usize>),
-
     // [4:]
     SliceFrom(usize),
+    // ==
+    Equal,
```
Comment (greyblake) [inline src/structs.rs]: In my opinion this comparison operators should separate entities (not Criterion)

**Comment by greyblake** [comment]
@MarcAntoine-Arnaud 

Hey, sorry again for delay. Finally I have time to take a look. Thanks again for the PR.

**Comment by greyblake** [comment]
Few notes regarding the PR:
* Please split PRs into smaller independent chunks (so it's easier to review). E.g. dependency update could be one PR, adding filter support - another.
* Use feature branches
* Please use 4 spaces for indent (instead of 2). It's a convention in Rust community

It's not a critic, it's just a feedback. Thanks again :+1:

**Comment by MarcAntoine-Arnaud** [comment]
You've right, I thinking just to push version update first, but I missed to switch to another branch.
I thinking to update with your comments but you've merged.

I will put 2 PR:
- fixing indentation
- spliting Criterion

**Comment by greyblake** [comment]
It's OK, just for the future)

**Pull request merged**